components/PostImage/PostImage.js

```diff
@@ -2,15 +2,50 @@ import styles from './PostImage.module.css';
 import cx from 'classnames';
 import React from 'react';
 
+const useElementOnScreen = (options) => {
+  const imgRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      setIsVisible(entry.isIntersecting);
```

Comment:
When you first run the app, all the posts you have will be observed but will be invisible, so you'll call this many times for no reason. I would do the following to only call it when needed:
```suggestion
      if(entry.isIntersecting !== isVisible) {
      	setIsVisible(entry.isIntersecting);
      }
```

Reply:
True.
I'll get it done.